src/dmd/backend/elfobj.d

```diff
@@ -1195,18 +1198,47 @@ void ElfObj_term(const(char)* objfilename)
             sechdr = &SecHdrTab[seg.SDrelidx];
             sechdr.sh_size = cast(uint)seg.SDrel.length();
             sechdr.sh_offset = foffset;
+
+            // sort the relocations by offset
             if (I64)
             {
                 assert(seg.SDrelcnt == seg.SDrel.length() / Elf64_Rela.sizeof);
-                debug for (size_t j = 0; j < seg.SDrelcnt; ++j)
-                {
-                    Elf64_Rela *p = (cast(Elf64_Rela *)seg.SDrel.buf) + j;
-                    if (ELF64_R_TYPE(p.r_info) == R_X86_64_64)
-                        assert(*cast(Elf64_Xword *)(seg.SDbuf.buf + p.r_offset) == 0);
-                }
+                extern (C) @trusted nothrow
+                    static int elf64_rel_fp(scope const(void*) e1,
+                                            scope const(void*) e2)
+                    {
+                        Elf64_Rela *r1 = cast(Elf64_Rela *)e1;
+                        Elf64_Rela *r2 = cast(Elf64_Rela *)e2;
+
+                        return cast(int)(r1.r_offset - r2.r_offset);
```

Comment:
I'm concerned about wraparound here when only the lower 32 bits are compared.

Reply:
Better if compare the full size of r_offset.

Reply:
Druntime likes to use this pattern I believe:
```suggestion
                        return (r1.r_offset > r2.r_offset) - (r1.r_offset < r2.r_offset); 
```

Reply:
I like this.